### ISSUE TITLE
CTA 1DC "all" index files & DataStore check

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1321]
@giovixo was the first to try to use the "all" index files from CTA 1DC, because he's analysing the Galactic center and wants to include runs from the "gps" and "gc" surveys:
https://github.com/gammasky/cta-analyses/tree/master/dc-1-checks/gc

There's a bug in `$CTADATA/index/all/obs-index.fits`: the reference time header keys haven't been added to the "all" index files, and since Gammapy accesses that at the moment when running a spectral analysis, it fails with a
```
KeyError: 'MJDREFI'
```
See traceback here: https://gist.github.com/cdeil/65a70e96e00021690a6832960db98580

A simple way to reproduce:
```
from gammapy.data import DataStore
data_store = DataStore.from_dir('$CTADATA/index/all')
print(data_store.obs(510000))
# -> KeyError: 'MJDREFI'
```

This is a reminder issue for myself to fix this here and update that file:
https://github.com/gammasky/cta-dc/blob/b74df3d5c9d5bccbea2ce229ffd1f0e0b8529700/data/cta_1dc_make_data_index_files.py#L289

But given that there have been issues with the index files before (see #1225), I really want to take the time now and add some validation / checker, that would have caught these mistakes.
Help welcome, if anyone is interested to work on the index files / datastore, let me know.

@giovixo or anyone that wants to use the "all" list - please use this to work around the issue for now:
```
from gammapy.data import DataStore
data_store = DataStore.from_dir('$CTADATA/index/all')
# A temp workaround for "all" index file bug.
# See https://github.com/gammapy/gammapy/issues/1320
meta = dict(MJDREFI=51544, MJDREFF=0.5, TIMEUNIT='s', TIMESYS ='TT', TIMEREF='LOCAL')
data_store.obs_table.meta.update(meta)
```

**Comment by cdeil** [comment]
@mackaiver mentioned in #1253 that there already is a [DataStore.check_integrity](http://docs.gammapy.org/dev/api/gammapy.data.DataStore.html#gammapy.data.DataStore.check_integrity) that is currently `NotImplemented` - just making this note for myself that this would be the place to implement the checks (or at least to expose them there, could call in other functions / classes of course).

**Comment by cdeil** [comment]
I now resolved this issue and added many checks in https://github.com/gammasky/cta-dc/blob/master/data/cta_1dc_make_data_index_files.py and https://github.com/gammasky/cta-dc/blob/master/data/cta_1dc_make_data_checks.py

I did not find new issues, with the index files, the index file from 8 month ago was good.

I did add units for a few columns in the observation index table, like "deg" for angles and "s" for times, but Gammapy seems to have worked without that, I don't think this will cause a change for anyone.

Sent the new tarball https://github.com/gammasky/cta-dc/commits/master/data/1dc/index.tar.gz with 
```
md5 == eb5efd5850bea200bf419f526b88aa07
```
to Jürgen for upload to Owncloud.

There is one potential new issue found concerning last event time and run stop time which don't completely agree. I've asked Jürgen about this via email.

Once this is done, the only remaining task is to leave one small comment in https://forge.in2p3.fr/boards/243/topics/2320 and close this issue.

Overall this shouldn't really affect analyses or warrant informing users to re-download the new version.

**Comment by cdeil** [comment]
This is done. I've also updated the `cta-1dc` dataset index files in gammapy-extra: https://github.com/gammapy/gammapy-extra/commit/ab8e319342ea1c63e040cccd66ad6b4f09246dca

**Comment by cdeil** [comment]
@robertazanin @facero @bkhelifi - Could you please do some testing here, and update the CTA DC-1 index files you have locally and re-run your tools checks analyses? Everything should work the same, results shouldn't change in any way. This is just to help check the new index files, to make sure there's no new issue.

How to download is described here
https://forge.in2p3.fr/projects/data-challenge-1-dc-1/wiki/Getting_data
but basically you need to run these two commands in the right folder to update the files in-place (extracting the tarball will overwrite the old files):
```
wget https://owncloud.cta-observatory.org/remote.php/webdav/1dc/index.tar.gz --user=<user> --ask-password
tar xfvz index.tar.gz
```

To check if you have the new index files, check the MD5 checksum, it should be this:
```
$ md5 index/all/obs-index.fits.gz 
MD5 (index/all/obs-index.fits.gz) = fecfe1828784f1e6f8d9e94acab1481a
```

Thanks!
